### ISSUE TITLE
[Feature Request]: Range Slider - Handle Tooltip Hovers

### ISSUE THREAD

**Issue by KevinCamelo** [carbon-design-system/carbon#14549]
### The problem

The range slider component currently relies on the user to review the left min or right max value to determine what number is being currently selected. Based on a conversation with the[Design Systems Adoption Guild](https://github.com/carbon-design-system/carbon/issues/6337#issuecomment-1682494447) and the Carbon team, tooltip indicators would be a welcome addition to the pattern. 

### The solution

Upon hover on the minimum or maximum value handle, show a tooltip that displays the number that corresponds to the active handle. 

![rangehover](https://github.com/carbon-design-system/carbon/assets/29249866/f3c22fc6-3186-4ede-a218-b3fd9779bc3b)

### Examples

| Example | Source | 
| --- | --- | 
| <img width="890" alt="Material" src="https://github.com/carbon-design-system/carbon/assets/29249866/0160c5a2-9aee-45f2-b18a-2ea442a7809c">| https://m3.material.io/components/sliders/specs |
| <img width="805" alt="Flutter" src="https://github.com/carbon-design-system/carbon/assets/29249866/fd858e33-2f62-4781-b13e-f9d7d885b427">| https://api.flutter.dev/flutter/material/RangeSlider-class.html|

**Audit link:** https://app.mural.co/t/cloudappservicesdesign3998/m/cloudappservicesdesign3998/1692204645335/c54284851ac0a172eef2d7b2e6bd918416cd738e?sender=ua50b53708bb0fd6ce5c12121

### Application/PAL

Carbon for Cloud

### Business priority

Low Priority = release date is not dependent on fix or not upcoming

### Available extra resources

None, but work has been done on the base version of the Range Slider component via #6337. What needs to be added is the tooltip interaction.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by sstrubberg** [comment]
@m4olivei any chance you can add this spec to #14297 ?

Let me know!

**Comment by m4olivei** [comment]
> @m4olivei any chance you can add this spec to https://github.com/carbon-design-system/carbon/pull/14297 ?

Hey @sstrubberg 👋 . @KevinCamelo and I agreed that this could happen as a follow up to #14297 . Is there something that's changed such that this is required?

**Comment by sstrubberg** [comment]
We were chatting about it in our backlog grooming call this morning and thought it would be worthwhile to see if you _could_ bring it in to the current work. No worries if not. We can always tackle it later. Do you foresee a lot of changes to incorporate it?

**Comment by m4olivei** [comment]
@sstrubberg I don't think it would be a lot of changes. Mainly I'm hoping to get what we have committed to avoid continual maintenance of merge conflicts. Already there is a non-trivial merge conflict that I need to resolve on account of upstream changes to the Slider component that were merged recently. It's also true that the team I work on doesn't have a pressing need for this component, so I get allocated to other projects ahead of this. Finding time to work on this is a challenge.

**Comment by sstrubberg** [comment]
Totally get it. Get your stuff merged. Based on your availability we'll tackle this next.

**Comment by m4olivei** [comment]
Hey all 👋 . I have time to pick this up in our current Sprint cycle. I've been experimenting locally. A couple of questions for you all:

* Would we also like to have tooltips for the single handle slider as well? 
* If yes, I'd propose that it would be opt-in for backward compatibility, adding a `tooltips` prop to turn on the functionality. Although maybe the argument for tooltips is strong enough here to have it be the default for single and double handle?
* I presume we want to use the [Carbon tooltip component](https://react.carbondesignsystem.com/?path=/story/components-tooltip--default) here? The Carbon tooltip component comes with more padding than is depicted here. Are we OK with the [Carbon tooltips](https://react.carbondesignsystem.com/?path=/story/components-tooltip--playground&args=align:top;label:16) as is?

![image](https://github.com/carbon-design-system/carbon/assets/191049/880401d6-14ef-4c9a-aa0d-cde849c57722)

**Comment by KevinCamelo** [comment]
@m4olivei Oh great Matt, excited to see! Chiming in on the one design related question, although I'm a fan of the making this a default option as that was the original intent.

> I presume we want to use the [Carbon tooltip component](https://react.carbondesignsystem.com/?path=/story/components-tooltip--default) here? The Carbon tooltip component comes with more padding than is depicted here. Are we OK with the [Carbon tooltips](https://react.carbondesignsystem.com/?path=/story/components-tooltip--playground&args=align:top;label:16) as is?

I wonder if we can use the definition tooltip here for the slightly tighter padding?  Is that against policy Carbon design squad? @laurenmrice

**Comment by m4olivei** [comment]
> I'm a fan of the making this a default option as that was the original intent.

On second thought, I'm up for that as well. If we can agree on default to tooltips for single and double handle slider, that would be great. Maybe we still have a prop to allow consumers to disable tooltips in case it conflicts in some way with their existing application. Would love if Carbon maintainers would chime in on that.

**Comment by m4olivei** [comment]
I opened a PR with progress so far:

https://deploy-preview-15129--v11-carbon-react.netlify.app/?path=/story/components-slider--two-handle-slider

I've assumed here that we turn on tooltips by default for both slider varients. There is also no way to opt out at present, I can add that in as we refine the open questions here. Look forward to any feedback.

**Comment by mbgower** [comment]
What is the rationale of having the hover? It seems wholly redundant to the input field.

**Comment by mjabbink** [comment]
@KevinCamelo What you describe as the “problem” is not a problem. It’s also not an accessibility issue, but rather an information overload, heavy-handed, and what I call “belt and suspenders”—all things we try avoid. 

The animated example visualizes more of a problem—too much information for no “real” benefit. It’s what I call forced benefit. Please closely examine the amount of information/numbers displayed in the tiny area below. I see a lot of numbers and repeating on top of that.


<img width="395" alt="Screenshot 2023-11-09 at 9 50 35 AM" src="https://github.com/carbon-design-system/carbon/assets/32881239/30bb6c7a-d480-40fc-8f0b-b0fd01a301cd">


<img width="190" alt="Screenshot 2023-11-09 at 9 53 30 AM" src="https://github.com/carbon-design-system/carbon/assets/32881239/44e0d6ff-725f-47ee-8276-a4f84bcebacb">

<img width="206" alt="Screenshot 2023-11-09 at 9 53 43 AM" src="https://github.com/carbon-design-system/carbon/assets/32881239/f0f01ce1-658d-4ca6-add5-07427239c92f">



The examples you show do not have input fields with value, so in that case, it makes sense. **We will make that an option and NOT the option to do toggle tip with value and input with value.** 

We will add to our guidance to not use both simultaneously but to create an alternate approach, which we do think.

**Comment by KevinCamelo** [comment]
Hi all. Thanks for the notes here — the range slider squad met today to discuss our actions moving forward .

## TLDR:
- The range slider component will now use tooltips in an either/or scenario. If text-inputs are present tooltips will not be displayed. If no text-inputs are present, tooltips will be displayed on hover, focus, and selection.
- This logic will extend to the single slider. 
- A future enhancement is planned for changes to the way the line segment in a range slider is presented.

## Tooltips: 
Instead of adding tooltips alongside the text inputs, the decision for product teams will be either/or. 
- If a team decides to use the text-input, they will not get the tooltip hover. 
- If a team disables the text-inputs, the tooltips will appear. 

The tooltip style used will be the style provided by the`icon` variant of the tooltip. 

@m4olivei here is a diagram of the requested design changes.

<img width="1378" alt="Screenshot 2023-11-09 at 2 04 35 PM" src="https://github.com/carbon-design-system/carbon/assets/29249866/19845c7b-6275-4e79-b943-52c74997e90b">

## Line segment updates
Open issues mentioned during the call was a lack of clarity of when the line starts or ends and the difficulty in seeing the line-segment (that represents the range) behind the selected line.
-  An example was presented with using the `gray 30` token to make it easier to understand where the line segment starts and ends. 
- Michael Gower also shared an example of how Carbon Charts starts and ends its line segments. 
- This will be acted on in a future enhancement to this component.

**Comment by laurenmrice** [comment]
@KevinCamelo Yes, I sign off on this. Thank you for documenting what we talked about in our meeting.

**Comment by m4olivei** [comment]
Thanks for the detailed notes @KevinCamelo . I think I've covered off all the requirements [in the PR](https://github.com/carbon-design-system/carbon/pull/15129). There are detailed testing instructions in the PR, but in short, I've added a couple stories to highlight the tooltips use case:

* [Slider > Slider With Hidden Inputs](https://deploy-preview-15129--v11-carbon-react.netlify.app/?path=/story/components-slider--slider-with-hidden-inputs)
* [Slider > Two Handle Slider With Hidden Inputs](https://deploy-preview-15129--v11-carbon-react.netlify.app/?path=/story/components-slider--two-handle-slider-with-hidden-inputs)

Be sure to review other stories for regressions in both Slider and Tooltip components as there were changes necessary in both places to fulfill the requirements. 

Thanks!
